src/framework/cfgTools.ml

```diff
@@ -135,14 +135,35 @@ let () = Printexc.register_printer (function
     | _ -> None (* for other exceptions *)
   )
 
+(** Type of CFG "edges": keyed by 'from' and 'to' nodes,
+    along with the list of connecting instructions. *)
+module CfgEdge = struct
+  open Batteries
+
+  type t = node * MyCFG.edges * node
+
+  let equal = Tuple3.eq Node.equal (List.equal (Tuple2.eq CilType.Location.equal Edge.equal)) Node.equal
+  let hash =
+    (* map the nodes to their hash, and each (location, edge) pair to their respective hashes;
+       then we have a structure of primitive constructors and ints, and can use polymorphic hash *)
+    Tuple3.map Node.hash (List.map (Tuple2.map CilType.Location.hash Edge.hash)) Node.hash %> Hashtbl.hash
```

Comment:
Yes, full type needs to be given so `deriving` can find things but otherwise done:

```suggestion
  type t = Node.t * MyCFG.edges * Node.t [@@deriving eq, hash]
```

Also derived eq/hash for `edges` (also requires writing out type names).